takeshape/penny
# Sentry

### Screenshots

![Screen Shot 2022-06-17 at 11 18 16 AM](https://user-images.githubusercontent.com/211047/174355402-4b066b50-1834-421e-8799-c63ca5bb6407.png)

### Test Plan (Steps to test):

1. Log into [the sentry project](https://sentry.io/organizations/takeshape-deluxetm-sample-proj/issues/). I sent out invites but let me know if you didn't get it.
1. Pull down this branch
1. Either log in using the sentry cli or create this `.sentryclirc` in the project root with a sentry auth token:

```
[auth]
token=YOUR_TOKEN_HERE
```

1. Add these lines to your `.env.local` `NEXT_PUBLIC_SENTRY_DSN='https://0a35260d726441119b87252042309d38@o1291323.ingest.sentry.io/6512748'` `SENTRY_ORG='takeshape-deluxetm-sample-proj'`
`SENTRY_PROJECT='deluxe-sample-project'`

1. Add an error button somewhere convenient. I put it at the top of the contact page.

```tsx
<button
  type="button"
  onClick={() => {
    throw new Error('Sentry Frontend Error');
  }}
>
  Throw error
</button>
```

1. Run the client and click the button, then make sure the error shows up in Sentry (see screenshot). There are already some errors in there so check the timestamp to make sure it's yours.
1. Add an error to the API, specifically `src/pages/api/auth/[...nextauth].ts`. I replaced the final return with this:

```ts
throw new Error('Sentry API Error');
```

1. Refresh your browser tab, check for the error in the console, then make sure the error shows up in Sentry. There are already some errors in there so check the timestamp.

### Checklist:
